akka-persistence-query/src/main/scala/akka/persistence/query/EventEnvelope.scala

```diff
@@ -59,14 +62,49 @@ final class EventEnvelope(
   def this(offset: Offset, persistenceId: String, sequenceNr: Long, event: Any, timestamp: Long) =
     this(offset, persistenceId, sequenceNr, event, timestamp, None)
 
+  /**
+   * Scala API
+   */
+  @deprecated("Use metadata with metadataType parameter")
+  def eventMetadata: Option[Any] = {
+    // For backwards compatibility this will use the metadata that was added last (ReplicatedEventMetaData)
+    _eventMetadata match {
+      case Some(CompositeMetadata(entries)) => entries.headOption
+      case s @ Some(_)                      => s
+      case None                             => None
```

Comment:
nitpicking...


This will look better as

```suggestion
    _eventMetadata.collect {
      case CompositeMetadata(entries) => entries.head
      case s                          => s
    }
```

Assuming of course that a CompositeMetadata has always a head.

Reply:
ok, will change, and see if there are any more places like that. Will also add a validation in CompositeMetadata that it must not be empty